tests/unit/customizations/ecs/test_codedeployer.py

```diff
@@ -89,7 +89,7 @@ def test_get_create_deploy_request(self):
         self.assertEqual(actual_hash, self.deployer._get_appspec_hash())
 
     def test_get_appspec_hash(self):
-        appspec_str = json.dumps(self.deployer._appspec_dict)
+        appspec_str = json.dumps(self.deployer._appspec_dict, sort_keys=True)
```

Comment:
So instead of updating the actual deploy logic, let's just update the `TEST_APPSPEC` to use `OrderedDict`. You should be able import the `OrderedDict` class from `awscli.compat`. The reason I'm suggesting this instead is that if we start off with an ordered dict it should guarantee order when it gets dumped meaning we do not need to update the actual source code for the command  and make this contribution be strictly a test update.